### PR TITLE
Get docker container to working state

### PR DESCRIPTION
**Pull request by carkmorwin**
This includes:
- Using alpine as base instead of scratch
- Pulling in ca-certificates package

---

When attempting to use the timber docker container `timberio/agent:0.6.1`, I encountered this error:

```
$ docker run timberio/agent:0.6.1
docker: Error response from daemon: OCI runtime create failed: container_linux.go:296: starting container process caused "exec: \"/bin/timber-agent\": stat /bin/timber-agent: no such file or directory": unknown.
```

This was due to pathing issue, with the timber agent build artifact being copied to `/timber-agent` but the entrypoint being `/bin/timber-agent`. This was fixed by updating the COPY command within the Dockerfile.

After this was fixed, I ran into another issue when attempting to ship logs:

```
2018/02/10 23:10:16 [ERR] POST https://logs.timber.io/frames request failed: 
Post https://logs.timber.io/frames: x509: failed to load system roots and no roots provided
```

After some googling, this appears to be caused by not having some root CAs certs available in the container, a by product of using  `scratch` as the base image. Turns out these certs are installed by default on most linux distros, so this is really a Docker specific problem.

Anyway, this was resolved by using alpine as the base image, and installing alpine's CA package, `ca-certificates` and insuring the certs are up to date. I went with alpine as it allowed for solving the problem, provides a base linux distro which can be attached to for debugging purposes, and retains a small container size, although not as small as before. 

```
$ docker images | grep timber
timber-alpine                                 latest              61680514a085        About a minute ago   14.7MB
timberio/agent                                0.6.1               653e4de7d2fd        4 weeks ago          8.3MB
```

I wanted to be thorough in this PR to highlight issues I ran into and minimal changes implemented to get things working. 

Lastly, the entire build output is being placed inside the container (binary, changelog, license, etc.). This aligns with the contents of the Github releases and thought these files should be included, but if not the `COPY` command in the Dockerfile can be changed to only include the bin. Also, the directory name was explicitly included (`timber-agent`) to avoid confusion when looking at the entrypoint (`/timber-agent/bin/timber-agent`).

Let me know if any other changes need to be made.

### CODE REVIEW

**Comment by binarylogic** [comment]
Awesome! I'm 👍 on the changes. @DavidAntaramian did the initial work on the `Dockerfile` so I'd like to get his review. I know originally we went with alpine and then switched to scratch based on a recommendation of a friend.

File: Dockerfile [hunk 1]
```diff
@@ -1,7 +1,12 @@
-FROM scratch
+FROM alpine
 
 ARG version
 
-COPY ./build/timber-agent-${version}-linux-amd64/ /
+RUN apk --update upgrade && \
+    apk add ca-certificates && \
+    update-ca-certificates && \
+    rm -rf /var/cache/apk/*
```
Comment (DavidAntaramian) [inline Dockerfile]: I think you can remove this line and change line 6 to be:

```
apk add --no-cache ca-certificates && \
```

File: Dockerfile [hunk 2]
```diff
@@ -1,7 +1,12 @@
-FROM scratch
+FROM alpine
```
Comment (DavidAntaramian) [inline Dockerfile]: Let's pin this to a specific version of alpine to reduce build inconsistencies.

**Comment by DavidAntaramian** [comment]
As a note for the future: This increases the size of the Docker image by about 9MB (total is 14.7MB now). That's still slim by current standards, but we may want to investigate just copying in the Mozilla CA Certificate Store. Since our binary is statically compiled, it doesn't need the additional resources provided by Alpine.

**Comment by carkmorwin** [comment]
@DavidAntaramian thanks for the feedback, and good points. Seems this is an issue that other people have solved via multi-stage builds, so I updated the Dockerfile to grab the CA certs in alpine and then only copy the necessary files to the scratch image. The new container built via this method worked as expected, and was closer to the original container size. Also pinned alpine to the latest current version.

```
$ docker images | grep timber
timber-alpine       latest              1e4bf04b2b0c        2 minutes ago       16.8MB
timber-multistage   latest              21c4581edd84        13 minutes ago      8.73MB
timberio/agent      0.6.1               653e4de7d2fd        4 weeks ago         8.3MB
```

Not sure why rebuilding the alpine image added 2MBs, but don't think it matters at this point.

**Comment by carkmorwin** [comment]
@DavidAntaramian Sorry for the delay on this. I added the comment to the top of the Dockerfile regarding the Mozilla certs and licensing as well as squashed the previous commits.

I took my best shot at the licensing and commit language but if any of that needs to change just let me know.

**Pull request merged**